server/methods/removeRoomLeader.js

```diff
@@ -0,0 +1,67 @@
+Meteor.methods({
+	removeRoomLeader(rid, userId) {
+		check(rid, String);
+		check(userId, String);
+
+		if (!Meteor.userId()) {
+			throw new Meteor.Error('error-invalid-user', 'Invalid user', {
+				method: 'removeRoomLeader'
+			});
+		}
+
+		if (!RocketChat.authz.hasPermission(Meteor.userId(), 'set-leader', rid)) {
+			throw new Meteor.Error('error-not-allowed', 'Not allowed', {
+				method: 'removeRoomLeader'
+			});
+		}
+
+		const user = RocketChat.models.Users.findOneById(userId);
+
+		if (!user || !user.username) {
+			throw new Meteor.Error('error-invalid-user', 'Invalid user', {
+				method: 'removeRoomLeader'
+			});
+		}
+
+		const subscription = RocketChat.models.Subscriptions.findOneByRoomIdAndUserId(rid, user._id);
+
+		if (!subscription) {
+			throw new Meteor.Error('error-user-not-in-room', 'User is not in this room', {
+				method: 'removeRoomLeader'
+			});
+		}
+
+		if (Array.isArray(subscription.roles) === true && subscription.roles.includes('leader') === false) {
+			throw new Meteor.Error('error-user-already-leader', 'User is not a leader', {
```

Comment:
Sorry, I meant just the error string, since the error is 'error-user-already-leader' and the message is 'User is not a leader', I think it should be something like 'User is already a leader'